config/config.yaml

```diff
@@ -250,3 +250,7 @@ assets:
             - decoders/pdfjs/jpg.js
             - decoders/rii-mango/lossless-min.js
             - dist/dwv.min.js
+    calculate-mosteller-formula:
+         basePath: '%webroot%/library/'
+         script: js/calculateBodySurfaceAreaMosteller.js
+         autoload: false
```

Comment:
just a minor change in path (js to path rather than the script) and remove the autoload (it defaults to false so no need to include it):
```
    calculate-mosteller-formula:
         basePath: '%webroot%/library/js/'
         script: calculateBodySurfaceAreaMosteller.js
```

Reply:
Regarding use, any places where we can use this in the codebase?

Reply:
one more thing that I completely forgot about. We shouldn't have any caps in path/filenames (except in special case of class name), so we should rename the filename to all lower caps, such as calculate-bsa-mosteller.js

Reply:
Okay, I made the fix in the main branch.

Reply:
hi @RachelElli, were you thinking the vitals form would be a good place to display this or is there another place?

Reply:
Eventually I think it would be nice to have a drop down menu there or elsewhere to do somewhat common/tedious calculations.  I have started to work on Mean Arterial Pressure(MAP).  Ideal body weight might also be useful along with adjusted body weight for dosing calculations.  I changed the filename and and updated the yaml file.

Reply:
ok, sure but should try to implement now or else it's just not useful

Reply:
That's fair.  I figured I'd start by adding the formulas first in case other people want to use them.  I recall that someone had asked on the forums before to be able to automate calculating BSA.

Reply:
prob be nice to have in vitals like how bmi is autocalculated if height and weight are input, will take some lifting but not too heavy

Reply:
I'd be hesitant to store it in the vitals table since it's not a part of the fhir v4 vitals panel
http://hl7.org/fhir/R4/observation-vitalsigns.html

which might make implementing fhir for meaningful use a little more convoluted since it's part of the US core requirements.
https://www.hl7.org/fhir/us/core/

Reply:
we can store things in our tables without worrying about 1 to 1 mapping to fhir

Reply:
This could be done without even storing anything. In forms report.php for vitals, could show BSA when there is a weight and height (this would take a php function rather than a js function to do). And then in the vitals form, could calculate bsa for the prior entries (via php function) and then calculate the items as inputted in the current entry (via the js function).

Something neat to consider also would be collecting the most recent weight, height, and calculated from bsa from that in the vitals widgets on the patient summary page (it now just shows the last vitals entry form which may not contain height or weight).

Reply:
So would it be preferable that I implement formulas/conversions in php over javascript in general?  I assumed javascript would be better because php can initiate javascript and javascript can access javascript in other .js files after importing.

Reply:
In may cases, javascript is what you'd be using for calculators etc, but in the case of supporting in vitals form is where things get more complicated and need to support both.
1. the javascript function will have you covered on client side javascript code. For example, when a user places or changes data in a weight input and height input then, the javascript function would autocalculate and display the bsa.
2. the php function will cover server side code. For example, when the report.php is called to show the vitals form, then it is calculated on server if both a weight and height exist.